routes/cube_routes.js

```diff
@@ -1468,11 +1468,11 @@ function writeCard(res, card, maybe) {
   res.write(`"${name}",`);
   res.write(`${card.cmc},`);
   res.write(`"${card.type_line.replace('—', '-')}",`);
-  res.write(`${card.colors.join('')},`);
+  res.write(`${(card.colors || []).join('')},`);
   res.write(`"${carddb.cardFromId(card.cardID).set}",`);
   res.write(`"${carddb.cardFromId(card.cardID).collector_number}",`);
-  res.write(`${card.rarity},`);
-  res.write(`${card.colorCategory},`);
+  res.write(`${card.rarity && card.rarity !== 'undefined' ? card.rarity : rarity},`);
```

Comment:
```suggestion
  res.write(`${card.rarity && card.rarity !== undefined ? card.rarity : rarity},`);
```

Reply:
No, we need to check for the string undefined, that was the issue. Somehow this field has gotten into this bad state